### ISSUE TITLE
the # symbol inside a text breaks the parser

### ISSUE THREAD

**Issue by remixer-dec** [cheeriojs/cheerio#1075]
This code works fine
```javascript
const $ = require('cheerio');     
console.log($('b','test<b>123</b>'))
```

And this one returns nothing (except the options)
```javascript
const $ = require('cheerio');       
console.log($('b','test#1<b>123</b>'))
```

**Comment by gombosg** [comment]
I actually found the same behaviour with for example the '<' symbol. I know it's not safe HTML (supposed to be `&gt;`) but for example, jQuery still works with it.

**Comment by fb55** [comment]
Fixed by #985
